packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -110,8 +110,25 @@ export const Table = memo((props: Props) => {
     [data, width, columnMinWidth, footerItems, hasNestedData, isCountRowsSet]
   );
 
+  const toggleAllRowsExpandedRef = useRef<(value?: boolean) => void>();
+
   // Internal react table state reducer
-  const stateReducer = useTableStateReducer(props);
+  const stateReducer = useTableStateReducer({
+    ...props,
+    onSortByChange: (state) => {
+      // Collapse all rows. This prevents a known bug that causes the size of the rows to be incorrect due to
+      // using `VariableSizeList` and `useExpanded` together.
+      if (toggleAllRowsExpandedRef.current) {
+        toggleAllRowsExpandedRef.current(false);
+      } else {
+        console.error('toggleAllRowsExpandedRef.current is undefined');
```

Comment:
This is a bit weird. Like if this happens, is it a programming error? If not can we recover? It's not clear from this if this is something that user/we can ignore or something to fix in the future.